lib/services/common.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import "github.com/gravitational/teleport/api/types"
+
+// MarshalFunc is a type signature for a marshaling function.
+type MarshalFunc[T types.Resource] func(T, ...MarshalOption) ([]byte, error)
```

Comment:
I don't see any value in using generics here. 

```
type MarshalFunc func(any, ...MarshalOption) ([]byte, error)
```

Reply:
Without the generic, the new service implementation was not happy, basically. Otherwise I wouldn't have bothered.